src/coreclr/vm/jitinterface.cpp

```diff
@@ -1401,10 +1411,17 @@ void CEEInfo::getFieldInfo (CORINFO_RESOLVED_TOKEN * pResolvedToken,
                 fieldAccessor = CORINFO_FIELD_STATIC_SHARED_STATIC_HELPER;
 
                 pResult->helper = getSharedStaticsHelper(pField, pFieldMT);
-
-#ifdef HOST_WINDOWS
-#ifndef TARGET_ARM
-                // For windows, we convert the TLS access to the optimized helper where we will store
+#if defined(TARGET_ARM)
+                // Optimization is disabled for linux/windows arm
+#elif !defined(TARGET_WINDOWS) && defined(TARGET_X86)
+                // Optimization is disabled for linux/x86
+#elif defined(TARGET_ALPINE_LINUX)
+                // Optimization is disabled for linux/alpine
+#elif defined(FEATURE_READYTORUN)
+                // Optimization is disabled for R2R
```

Comment:
```suggestion
```
`FEATURE_READYTORUN` means that the runtime supports R2R. It is always defined when compiling the shipping runtimes. It does not indicate whether the JIT is producing R2R code currently.

I think this means that the optimization is always disabled in the current version of the change.

Reply:
ah I see. let me see if there is a better way we have to check if we are producing for R2R.